common/metrics/defs.go

```diff
@@ -1692,68 +1628,55 @@ var MetricDefs = map[ServiceIdx]map[int]metricDefinition{
-		ArchiverPumpedNotEqualHandledCount:                     {metricName: "archiver_pumped_not_equal_handled"},
-		ArchiverHandleAllRequestsLatency:                       {metricName: "archiver_handle_all_requests_latency"},
-		ArchiverWorkflowStoppingCount:                          {metricName: "archiver_workflow_stopping"},
-		ArchiverClientSendSignalFailureCount:                   {metricName: "archiver_client_send_signal_error"},
-		TaskProcessedCount:                                     {metricName: "task_processed", metricType: Gauge},
-		TaskDeletedCount:                                       {metricName: "task_deleted", metricType: Gauge},
-		TaskListProcessedCount:                                 {metricName: "tasklist_processed", metricType: Gauge},
-		TaskListDeletedCount:                                   {metricName: "tasklist_deleted", metricType: Gauge},
-		TaskListOutstandingCount:                               {metricName: "tasklist_outstanding", metricType: Gauge},
-		StartedCount:                                           {metricName: "started", metricType: Counter},
-		StoppedCount:                                           {metricName: "stopped", metricType: Counter},
-		ExecutorTasksDeferredCount:                             {metricName: "executor_deferred", metricType: Counter},
-		ExecutorTasksDroppedCount:                              {metricName: "executor_dropped", metricType: Counter},
-		BatcherProcessorSuccess:                                {metricName: "batcher_processor_requests", metricType: Counter},
-		BatcherProcessorFailures:                               {metricName: "batcher_processor_errors", metricType: Counter},
+		ReplicatorMessages:                       {metricName: "replicator_messages"},
+		ReplicatorFailures:                       {metricName: "replicator_errors"},
+		ReplicatorMessagesDropped:                {metricName: "replicator_messages_dropped"},
+		ReplicatorLatency:                        {metricName: "replicator_latency"},
+		ESProcessorRequests:                      {metricName: "es_processor_requests"},
+		ESProcessorRetries:                       {metricName: "es_processor_retries"},
+		ESProcessorFailures:                      {metricName: "es_processor_errors"},
+		ESProcessorCorruptedData:                 {metricName: "es_processor_corrupted_data"},
+		ESProcessorProcessMsgLatency:             {metricName: "es_processor_process_msg_latency", metricType: Timer},
+		IndexProcessorCorruptedData:              {metricName: "index_processor_corrupted_data"},
+		IndexProcessorProcessMsgLatency:          {metricName: "index_processor_process_msg_latency", metricType: Timer},
+		ArchiverNonRetryableErrorCount:           {metricName: "archiver_non_retryable_error"},
+		ArchiverHistoryMutatedCount:              {metricName: "archiver_history_mutated"},
+		ArchiverTotalUploadSize:                  {metricName: "archiver_total_upload_size", metricType: Timer},
```

Comment:
I keep the metrics for mutated history and total upload size which I think should be useful for other implementations.